From 89871ab5eee02795992990156303a85ef7ce39a4 Mon Sep 17 00:00:00 2001
From: jimmyx <jimmyx@supermicro.com>
Date: Fri, 18 Aug 2017 15:40:39 -0700
Subject: [PATCH 7/8] fix wrong slot number issue. #371

---
 platforms/astbmc/p9dsu.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/platforms/astbmc/p9dsu.c b/platforms/astbmc/p9dsu.c
index 8b6799c..4f3eecc 100644
--- a/platforms/astbmc/p9dsu.c
+++ b/platforms/astbmc/p9dsu.c
@@ -28,12 +28,12 @@ static const struct slot_table_entry p9dsu_uio_plx_down[] = {
 	{
 		.etype = st_builtin_dev,
 		.location = ST_LOC_DEVFN(0x1,0),
-		.name = "PLX switch",
+		.name = "UIO Slot0",
 	},
     	{
 		.etype = st_pluggable_slot,
 		.location = ST_LOC_DEVFN(0x8,0),
-		.name = "UIO Slot0",
+		.name = "PLX switch",
 	},
 	{
 		.etype = st_builtin_dev,
@@ -57,12 +57,12 @@ static const struct slot_table_entry p9dsu_wio_plx_down[] = {
 	{
 		.etype = st_builtin_dev,
 		.location = ST_LOC_DEVFN(0x1,0),
-		.name = "PLX switch",
+		.name = "WIO Slot1",
 	},	
     	{
 		.etype = st_pluggable_slot,
 		.location = ST_LOC_DEVFN(0x8,0),
-		.name = "WIO Slot1",
+		.name = "PLX switch",
 	},
 	{
 		.etype = st_pluggable_slot,
-- 
2.5.0.windows.1

