From 8d6e90e50aa0ed2e250147098451adfdedcf6a16 Mon Sep 17 00:00:00 2001
From: jim <jim.yuan@supermicro.com>
Date: Fri, 14 Jul 2017 18:40:11 -0700
Subject: [PATCH] Cut down the large number of traces from PNOR MBOX

---
 src/usr/pnor/pnor_mboxdd.C | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/usr/pnor/pnor_mboxdd.C b/src/usr/pnor/pnor_mboxdd.C
index 4757ff1..088fc97 100644
--- a/src/usr/pnor/pnor_mboxdd.C
+++ b/src/usr/pnor/pnor_mboxdd.C
@@ -250,7 +250,7 @@ errlHndl_t PnorDD::writeFlash(const void* i_buffer,
                               size_t& io_buflen,
                               uint64_t i_address)
 {
-    TRACFCOMP(g_trac_pnor, ENTER_MRK"PnorDD::writeFlash(i_address=0x%llx)> ", i_address);
+    TRACDCOMP(g_trac_pnor, ENTER_MRK"PnorDD::writeFlash(i_address=0x%llx)> ", i_address);
     errlHndl_t l_err = NULL;
 
     uint32_t l_address = i_address & 0x0fffffff;
@@ -264,7 +264,7 @@ errlHndl_t PnorDD::writeFlash(const void* i_buffer,
         io_buflen = 0;
     }
 
-    TRACFCOMP(g_trac_pnor, EXIT_MRK"PnorDD::writeFlash(i_address=0x%llx)> io_buflen=%.8X", i_address, io_buflen);
+    TRACDCOMP(g_trac_pnor, EXIT_MRK"PnorDD::writeFlash(i_address=0x%llx)> io_buflen=%.8X", i_address, io_buflen);
 
     return l_err;
 }
@@ -432,7 +432,7 @@ errlHndl_t PnorDD::_writeFlash( uint32_t i_addr,
                                 size_t i_size,
                                 const void* i_data )
 {
-    TRACFCOMP(g_trac_pnor, ENTER_MRK"PnorDD::_writeFlash(i_addr=0x%.8X)> ", i_addr);
+    TRACDCOMP(g_trac_pnor, ENTER_MRK"PnorDD::_writeFlash(i_addr=0x%.8X)> ", i_addr);
     errlHndl_t l_err = NULL, l_flushErr = NULL;
 
     while (i_size)
@@ -502,7 +502,7 @@ errlHndl_t PnorDD::_readFlash( uint32_t i_addr,
                                size_t i_size,
                                void* o_data )
 {
-    TRACFCOMP(g_trac_pnor, "PnorDD::_readFlash(i_addr=0x%.8X)> ", i_addr);
+    TRACDCOMP(g_trac_pnor, "PnorDD::_readFlash(i_addr=0x%.8X)> ", i_addr);
     errlHndl_t l_err = NULL;
 
     while (i_size)
-- 
2.9.2.windows.1

